src/Metadata/Source/MysqlMetadata.php

```diff
@@ -244,36 +248,33 @@ protected function loadConstraintData($table, $schema)
 
         $realName = null;
         $constraints = [];
-        foreach ($results->toArray() as $row) {
-            if ($row['CONSTRAINT_NAME'] !== $realName) {
-                $realName = $row['CONSTRAINT_NAME'];
-                $isFK = ('FOREIGN KEY' == $row['CONSTRAINT_TYPE']);
-                if ($isFK) {
+        if ($results !== null) {
+            foreach ($results->toArray() as $row) {
+                if ($row['CONSTRAINT_NAME'] !== $realName) {
+                    $realName = $row['CONSTRAINT_NAME'];
+                    $isFK = ('FOREIGN KEY' === $row['CONSTRAINT_TYPE']);
                     $name = $realName;
-                } else {
-                    $name = '_laminas_' . $row['TABLE_NAME'] . '_' . $realName;
+                    $constraints[$name] = [
+                        'constraint_name' => $name,
+                        'constraint_type' => $row['CONSTRAINT_TYPE'],
+                        'table_name'      => $row['TABLE_NAME'],
+                        'columns'         => [],
+                    ];
+                    if ($isFK) {
+                        $constraints[$name]['referenced_table_schema'] = $row['REFERENCED_TABLE_SCHEMA'];
+                        $constraints[$name]['referenced_table_name']   = $row['REFERENCED_TABLE_NAME'];
+                        $constraints[$name]['referenced_columns']      = [];
+                        $constraints[$name]['match_option']       = $row['MATCH_OPTION'];
+                        $constraints[$name]['update_rule']        = $row['UPDATE_RULE'];
+                        $constraints[$name]['delete_rule']        = $row['DELETE_RULE'];
+                    }
                 }
-                $constraints[$name] = [
-                    'constraint_name' => $name,
-                    'constraint_type' => $row['CONSTRAINT_TYPE'],
-                    'table_name'      => $row['TABLE_NAME'],
-                    'columns'         => [],
-                ];
+                $constraints[$name]['columns'][] = $row['COLUMN_NAME'];
```

Comment:
this name will be empty if above case wont be triggered?)

Reply:
@tptrixtop possibly, however that potential case isn't new to this commit; it has existed for some time.
PhpStorm also warns me about line 272, saying `$isFK` is probably undefined:
```
if ($isFK) {
```

Reply:
looks like `$name` was initializing in else { case in previous version, in new version else case was removed and allows `$name` to be uninitialized or i am missing something